src/gui/entry/attachments/TextAttachmentsWidget.cpp

```diff
@@ -65,7 +65,15 @@ void TextAttachmentsWidget::updateWidget()
     }
 
     m_editWidget->openAttachment(m_attachment, m_mode);
-    m_previewWidget->openAttachment(m_attachment, attachments::OpenMode::ReadOnly);
+}
+
+void TextAttachmentsWidget::updatePreviewWidget()
+{
+    m_previewVisible = m_previewWidget->width() > 0;
```

Comment:
This `m_previewWidget->width() > 0` is repeated many times in the file. Maybe create a function for it?

Reply:
It's kind of crazy qt doesn't offer an `isCollapsed` function on the QSplitter. `isVisible` always returns true as well. Good idea on a defined function.